sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,422 @@
+    task_sequence<vectorSum> thirdQuarter;
+    task_sequence<vectorSum> fourthQuarter;
+    int quarterCount = kCount/4;
+    firstQuarter.async(in.data(), 0, quarterCount);
+    secondQuarter.async(in.data(), quarterCount, quarterCount);
+    thirdQuarter.async(in.data(), 2*quarterCount, quarterCount);
+    fourthQuarter.async(in.data(), 3*quarterCount, quarterCount);
+    *out = firstQuarter.get() + secondQuarter.get() +
+      thirdQuarter.get() + fourthQuarter.get();
+  }).wait();
+  ...
+}
+```
+
+== Definitions
+
+== `task_sequence` class
+
+[source,c++,linenums]
+----
+namespace sycl::ext::intel::experimental {
+
+template<auto& f, class propertiesT =
+   decltype(oneapi::experimental::properties{})>
+class task_sequence{};
+
+template <typename ReturnT, typename... ArgsT, ReturnT (&f)(ArgsT...),
+          class propertiesT>
+class task_sequence<f, propertiesT> {
+
+public:
+  task_sequence(const task_sequence &) = delete;
+  task_sequence &operator=(const task_sequence &) = delete;
+  task_sequence(task_sequence &&) = delete;
+  task_sequence &operator=(task_sequence &&) = delete;
+
+  task_sequence();
+
+  task_sequence(const propertiesT &);
+
+  void async(ArgsT... args);
+
+  ReturnT get();
+  
+  template <typename propertyT>
+  static constexpr bool has_property();
+  
+  template <typename propertyT>
+  static constexpr auto get_property();
+
+  ~task_sequence();
+};
+
+} // namespace experimental
+} // namespace intel
+} // namespace ext
+} // namespace sycl
+
+----
+
+`task_sequence` is a class template, parameterized by an `auto` reference to a
++Callable+ `f` and by a list of properties `propertiesT`.
+The +Callable+ `f` defines the asynchronous task to be
+associated with the `task_sequence`, and requiring an auto reference ensures
+that each `f` be statically resolvable at compile time.  Static resolvability by
+the compiler is desirable when compiling for spatial architectures as it can
+enable the generation of more efficient hardware. Furthermore, the partial
+specialization `typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...)`
+helps make it easier to write the class member functions (eg: `get()`).
+
+The `task_sequence` interface consists of two member functions, `async` and
+`get`. The `async` function asynchronously invokes `f`, and stores the return
+value into a FIFO queue upon completion of `f`. By calling async on a
+`task_sequence` more than once, the user implies that the invocations of `f` can
+be run in parallel, although the implementation is by no means obligated to do so.  
+
+The `async` function call is non-blocking in that it may return before the
+asynchronous `f` invocation completes executing, and potentially before `f` even
+begins executing (return from the `async` provides no implicit information on
+the execution status of `f`).
+
+The `get` member function retrieves the oldest result from this logical FIFO
+queue, and blocks (waits) until a result is available if none are available
```

Comment:
This combined with the sentence above:

> stores the return value into a FIFO queue upon completion of `f`

implies that values are returned in arbitrary order (as the programmer has no guarantee about the order in which the invoked tasks complete) as opposed to the order the task are launched in.  Is this really our intent?  I was under the impression that we'd get() in the same order the user async()ed so that users could reason about which invocation's result they are looking at.

Reply:
Looking back through our internal materials as we were developing this, the intent was always that the get()s would return results in the same order the tasks were async()ed so I think this is a mistake.

Reply:
Removed this line: "stores the return value into a FIFO queue upon completion of f"
Removed the mention of FIFO in other places in the doc.
Clearly specified that the results are retrieved in the same order as that of the `async` calls.